### DISCUSSION TITLE
elide.modelPackage

### DISCUSSION THREAD

**Discussion by white3km** [yahoo/elide#2556] [General]
Hello, I have a question about the elide.modelPackage setting.  I noticed it's used in the elide standalone version and that it's included in the spring boot example (https://github.com/yahoo/elide-spring-boot-example/blob/master/src/main/resources/application.yaml#L8) but I don't see where it actually gets passed in for the elide spring boot project (https://github.com/yahoo/elide/blob/master/elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/ElideConfigProperties.java).

I was hoping to use this setting to toggle between two different model packages (ex. alpha & beta). When elide.modelPackage=alpha then beta is ignored and not picked up by elide. Am I understanding this setting correctly? Would it work for my use case?  Also are you able to list multiple packages?

**Comment by aklish** [comment]
Spring boot has its own mechanism for controlling which entities/packages are scanned (so Elide spring boot leverages that instead of using the elide.modelPackage).   That property is defunct for spring boot and we should remove it from the example.

You can either control spring boot package scans using the `@EntityScan` annotation or by adding a persistence.xml file and creating your own EntityManagerFactory bean.  I'm not sure if there is a convenient way to control scanned packages through application.yaml (but I'm betting there is).

**Comment by moizarafat** [comment]
Something like this can be used `@EntityScan(basePackages = {"${path.scan}"})`. `path.scan` can be added in the yaml file.
